### DISCUSSION TITLE
How should plans work?

### DISCUSSION THREAD

**Discussion by Glench** [Glench/ExtPay#79] [Ideas]
Many of you have asked to have plans or multiple pricing options in ExtensionPay. I think this is a great idea and would love feedback. Could you share how you'd like plans to work in your extension? How many different plans are you thinking? Are there different payment frequencies between plans? Different feature sets between plans? How would you imagine a user chooses a plan — from in your extension or in the ExtensionPay payment screen itself? I'm looking for feedback to design everything from the ExtPay.js api to the payment screen, so your input is greatly appreciated.

**Comment by avi12** [comment]
Sadly I can't use ExtPay as it's relying on Stripe which isn't available in my country

**Reply by FrazerHainsworth** [reply]
Monthly and Anually coices.
or
Monthly and Lifetime

these options would be useful for my extention

**Comment by brismuth** [comment]
I've been thinking about adding plans. My main use case would be to support different feature sets between plans. I'd be fine with managing the plan selection UI in the extension itself, rather than having ExtensionPay provide that.

**Comment by Lotsofjoy** [comment]
Would have loved to have at least 2 tiers with both monthly and annual payment options.

**Comment by MikyW** [comment]
I would like to offer different payment frequency (monthly/yearly). It can be either in extension or in ExtPay but I assume that having it in extension would provide more flexibility for design and copy.

**Comment by sicunningham** [comment]
Having the ability to offer more subscription terms other than just 1 month or 12 months - some middle ground like every 3 months would be really helpful.

**Comment by burkybang** [comment]
My customers are mainly interested in having the option to pay monthly or annually. I have not thought about having tiers for different features, but that sounds cool.

**Comment by bluecaret** [comment]
First, thank you for providing ExtPay! Using Stripe directly has not panned out for me with the whole manifest v3 thing so I'm very happy this exists!

For different plans, I've been considering two different paths:

- A monthly and yearly subscription with a third "lifetime" one-time purchase. Pretty straight-forward here. With this option though I would want a way to ensure users cannot accidentally sign up for multiple subscriptions at the same time, maybe that could still be managed in my extension, not sure.
- No subscriptions, but instead more of a DLC model where users can buy different "products" to unlock specific features, along with a bundle purchase that unlocks multiple purchases at once. With this option, I could potentially see anywhere between 5-15 offerings for the user, and more as I add new options. The benefit here is I don't have to continuously pump out new features to justify the subscription, instead, when I feel like it I can release new features and users can unlock them a-la-carte.

I think I would prefer to go with the second option, but since I cannot do this right now I might be forced the other route for my new project as I can offer the monthly one right away and add in the others later when its supported, switching from monthly to individual feature unlocks will be a harder transition to make users do.

I would prefer to run the experience of users switching between plans in my extension. Keep the interface on ExtPay minimal to just the payments and logging in. This would help keep the branding and trust in place in my mind. As much as possible, I'd prefer to keep the user within the extension to manage their payments/subscriptions.

Thank you! I'm sure it's early early, but any thoughts on when something like this might see a release? (even if something vague like "next year"?)

**Reply by angeogawin** [reply]
I would be very interested by the "No subscriptions, but instead more of a DLC model " to give users possibility to 
buy different products.  
Many of my users dont want subscriptions(to access all the products), but are ready for one time payment for a specific product.

**Comment by benf2004** [comment]
The biggest thing is that customers should be automatically canceled from their monthly plan when they upgrade to yearly/lifetime. I don't think this should be too hard if you could be able to link multiple extension IDs together. You wouldn't have to change anything other than allowing us to link the plans together. Right now using my [work around](https://github.com/Glench/ExtPay/discussions/40) I have a warning dialogue to confirm a user has canceled their monthly plan when upgrading.

**Reply by benf2004** [reply]
Also, I still want coupon codes 😉

**Comment by limbuscode** [comment]
You should add the option to set different features for each plan. But in my particular case, I need one-time, monthly, and yearly plans with the same set.

**Comment by angeogawin** [comment]
It would be great to have the possibility to sell differents products with one-time payment(not subscription). Currently we can do this only for one product.

**Comment by ba32107** [comment]
My use-case: I want to offer 3 tiers to the user: a free tier, a pro tier and a premium tier. The premium would be most expensive and have all the features. I want both paid tiers to be monthly subscription based, with an option to buy a full year with a discount.

@Glench I know you are still in the planning phase, but when do you think this functionality will be released? I'm just wondering if I need to implement something custom [like this](https://github.com/Glench/ExtPay/discussions/40) or if I should just wait for the official support. I'm planning to launch my extension in a couple of months.

**Comment by ba32107** [comment]
> How would you imagine a user chooses a plan — from in your extension or in the ExtensionPay payment screen itself?

My 2 cents: can we have both? Offer an option to populate the payment screen with all this, but also the flexibility for us to implement custom stuff.

> How many different plans are you thinking? Are there different payment frequencies between plans? Different feature sets between plans?

The way I imagine this is: on the extensionpay.com dashboard, for a given extension, I'd like to be able to define multiple (as many as I want) plans. For each plan, I would give a unique name (or get an auto-generated ID), and I want to set payment options for each plan. These payment options would be the same as the one that applies to the extension today (so I'd like to set price, frequency, etc).

Then in my extension code, I'd like something like `getAllPlans` which would return all the defined plans, as well as a `getActivePlan` function. With these I can implement any feature check I want in my extension. I do not think extensionPay should know anything about the different features the different plans offer - it does not add anything useful. All I need to know is which plan is active.

As for payments, I can imagine the `openPaymentPage` taking a config object parameter, and in that object, I could either set a `showAllPlans` flag (which would let the user choose which plan to sign up for), or I could set a `forPlan` property with a plan ID, if I choose to implement the plan chooser myself (in this case, the payment page would open for that specific plan).

This is just a rough idea, I know there are many other things to consider (how does the user change from one plan to another, can we have multiple active plans, etc)

**Comment by BowTiedSwan** [comment]
Hi, just found this product yesterday while developing an extension so haven't tested it live. So far having a price for monthly vs a reduced subscription if the user pays for the year would be great. 
Then adding the possibility of extra tiers to allow for different functionality (for example if I'm using some NLP or third party apis that are expensive to run, I'd like to have some users limited to X amount of requests/day/month and another tier where the limit is higher)

**Comment by davidgarry1** [comment]
I would just like to second that the most requested feature I get is to offer an annual *and* a monthly option. Currently I only do annual, and I think adding monthly at a slightly increased cost would be awesome.

I would strongly prefer if the ExtensionPay frame let's the user choose between the 2 options, and if the extension creators don't have to do special handling for the different cases (eg that "paid" boolean will still work as it currently does) if we don't want to. Basically just the same as it currently is, but with the added option of users being charged monthly or annually as they please, and switching between them as they please, entirely within the ExtensionPay window. This would be the perfect ideal situation for me.

**Comment by pizzaisdavid** [comment]
As an example, imagine three tiers:

The free plan - has a basic set of features.
The 5 dollar a month plan - has some more features.
The 15 dollar a month plan - has a lot more features.

**Comment by FrazerHainsworth** [comment]
Monthly and Anually coices.
or
Monthly and Lifetime

these options would be useful for my extention

**Comment by surtic86** [comment]
You should have the Option to make 3 or 6 Months plans... since it gets you less Fees on Creditcards and so on. Also you can then offer it a bit cheaper then 1month but you still look often better to have sold this then 1 month

**Comment by BowTiedSwan** [comment]
@Glench Updates on this? Thank you ser!

**Comment by bluecaret** [comment]
@Glench sorry to bug, but do you have any news at all on this feature? Thank you for your work!

**Comment by Glench** [comment]
We're actively working on this now! We won't share deadlines, but it's coming!

**Reply by bluecaret** [reply]
Thank you for the reply, glad to know it's still happening! Was getting worried it wasn't :)

**Reply by amonam** [reply]
Do you have any updates? Is it expected to publish within a month or a year? Just provide us with a rough estimate, please.

**Comment by MaritJoustra** [comment]
Thank you for offering ExtPay! 

We are thinking about having 3 types of payment; one time, monthly, yearly. Most flexible would be to be able to register all types on  ExtensionPay and then in our extensions choose which one to use. So for extension A I loop through ExtensionPay id's monthly and yearly to check if paid; if not paid offer options to choose mothly or yearly. For extension B we want to use either one time or monthly, so then loop through id's one time and monthly; if not paid then offer options to choose one time of monthly payment. Thank you for your time working on mulitple pricing!

**Comment by jtoelle** [comment]
You give away a free version to gain traction with the following upgrade options to a Pro version.

The 3 options to upgrade to a Pro version would be:
Monthly
Yearly
Lifetime

There should also be a coupon option to give the user

**Reply by pizzaisdavid** [reply]
I speculate lifetime plans don't make so much business sense for @Glench , as it is a onetime revenue and then potentially infinite ongoing expense.

**Reply by burkybang** [reply]
Lifetime doesn't make a lot of sense for my extension either for the same reason.

**Reply by jtoelle** [reply]
Lifetime is already an option in Extensionpay and makes a ton of sense, it's a great way to generate a massive amount of income upfront.

Appsumo makes millions per year offering the lifetime option on software.

**Reply by pizzaisdavid** [reply]
@jtoelle ah, you're right. My mistake. Sorry. I checked this before posting but on an existing subscription product. It looks like once a product is set as a subscription it only shows the option for monthly or yearly.

**Reply by takkbbl** [reply]
A coupon option would be amazing!
